sql/2021/css/content_values.sql

```diff
@@ -0,0 +1,63 @@
+#standardSQL
+# Most popular `content` property values.
+# Combines hex values together to reduce duplication (used by icon fonts).
+CREATE TEMP FUNCTION getContentStrings(css STRING) RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+  var reduceValues = (values, rule) => {
+    if ('rules' in rule) {
+      return rule.rules.reduce(reduceValues, values);
+    }
+    if (!('declarations' in rule)) {
+      return values;
+    }
+    return values.concat(rule.declarations.filter(d => d.property.toLowerCase() == 'content').map(d => d.value));
+  };
+  var $ = JSON.parse(css);
+  return $.stylesheet.rules.reduce(reduceValues, []);
+} catch (e) {
+  return [];
+}
+''';
+
+SELECT
+  *
+FROM (
+  SELECT
+    client,
+    IF(
+      REGEXP_CONTAINS(content, r'[\'"]\\[ef][0-9a-f]{3}[\'"]'), '"\\f000"-like',
+    IF(
+      REGEXP_CONTAINS(content, r'[\'"]\\[a-f0-9]{4}[\'"]'), '"\\hex{4}"-like', content)) AS content,
+    COUNT(DISTINCT page) AS pages,
+    total_pages,
+    COUNT(DISTINCT page) / total_pages AS pct_pages,
+    COUNT(0) AS freq,
+    SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
+    COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client) AS pct
+  FROM (
+    SELECT
+      client,
+      COUNT(DISTINCT page) AS total_pages
+    FROM
+      `httparchive.almanac.parsed_css`
+    WHERE
+      date = '2021-07-01'
+    GROUP BY
+      client)
+  JOIN
+    `httparchive.almanac.parsed_css`
+  USING
+    (client),
+    UNNEST(getContentStrings(css)) AS content
+  WHERE
```

Comment:
Potential UNNEST count issue.

Reply:
I think this is WAI: count of pages uses `DISTINCT page` and count of content items should account for each UNNESTed value.

Reply:
Could you double check this one? I’m less worried about the distinct page and more worried about this line (and the pct line based on it):

```sql
SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
```

This has potential to be a huge increase on total number.

Unresolving for now.

Reply:
Yeah I think it's ok. The total should be counting all `content` values. And the `WHERE pages >= 1000` is in a wrapped outer query that won't affect the total.

Reply:
An you’ve a separate “total_pages” column above which is calced with a sub query, so this one isn’t a total pages. Ok that makes sense. I think :-)